src/Service/EolValidator.php

```diff
@@ -0,0 +1,163 @@
+
+    /**
+     * @var FileList
+     */
+    private $fileList;
+
+    /**
+     * @var File
+     */
+    private $file;
+
+    /**
+     * @var ServiceFactory
+     */
+    private $serviceFactory;
+
+    /**
+     * @var array
+     */
+    private $eolConfigs;
+
+    /**
+     * @var array
+     */
+    private $services = [
+        ServiceInterface::NAME_PHP,
+        ServiceInterface::NAME_ELASTICSEARCH,
+        ServiceInterface::NAME_RABBITMQ,
+        ServiceInterface::NAME_REDIS,
+        ServiceInterface::NAME_DB
+    ];
+
+    /**
+     * @param FileList $fileList
+     * @param ServiceFactory $serviceFactory
+     */
+    public function __construct(
+        FileList $fileList,
+        File $file,
+        ServiceFactory $serviceFactory
+    ) {
+        $this->fileList = $fileList;
+        $this->file = $file;
+        $this->serviceFactory = $serviceFactory;
+    }
+
+    /**
+     * Validate the EOL of a given service and version by error level.
+     *
+     * @return array
+     * @throws FileSystemException
+     * @throws ServiceMismatchException
+     */
+    public function validateServiceEol(): array
+    {
+        $errors = [];
+
+        foreach ($this->services as $serviceName) {
+            $service = $this->serviceFactory->create($serviceName);
+            $serviceVersion = $service->getVersion();
+
+            if ($validationResult = $this->validateService($serviceName, $serviceVersion)) {
+                $errorLevel = array_key_first($validationResult);
+                $errors[$errorLevel][] = $validationResult[$errorLevel];
+            }
+        }
+        return $errors;
+    }
+
+    /**
+     * Validates a given service and version.
+     *
+     * @param string $serviceName
+     * @param string $serviceVersion
+     * @return array
+     * @throws FileSystemException
+     */
+    public function validateService(string $serviceName, string $serviceVersion) : array
+    {
+        $serviceConfigs = $this->getServiceConfigs($serviceName);
+
+        $versionConfigs = array_filter($serviceConfigs, function ($v) use ($serviceVersion) {
+            return Semver::satisfies($serviceVersion, sprintf('%s.x', $v['version']));
+        });
+
+        if (!$versionConfigs || $versionConfigs[array_key_first($versionConfigs)]['eol'] === null) {
+            return [ValidatorInterface::LEVEL_WARNING => sprintf(
+                'Unknown or invalid EOL defined for %s %s',
+                $serviceName,
+                $serviceVersion
+            )];
+        }
+
+        $eolDate = Carbon::createFromTimestamp($versionConfigs[array_key_first($versionConfigs)]['eol']);
+
+        if (!$eolDate->isFuture()) {
+            return [ValidatorInterface::LEVEL_WARNING => sprintf(
+                '%s %s has passed EOL (%s).',
+                $serviceName,
+                $serviceVersion,
+                date_format($eolDate, 'Y-m-d')
+            )];
+        } else if ($eolDate->isFuture()
```

Comment:
Please use `elseif` instead of `else if`.
PSR-2: `The keyword elseif SHOULD be used instead of else if so that all control keywords look like single words.`

And please fix the formating of multiline conditions.

```
...
} elseif ($eolDate->isFuture()
        && $eolDate->diffInMonths(Carbon::now()) <= self::NOTIFICATION_PERIOD
) {
... 
```